week 2/script.js

```diff
@@ -0,0 +1,219 @@
+let currencyRates = {
+    timestamp: 1519296206,
+    base: "EUR",
+    date: "2021-03-17",
+    rates: {
+        USD: 1.23396
+    }
+};
+
+class CurrencyRate {
+    constructor(code, rate, date) {
+        this.code = code;
+        this.rate = rate;
+        this.date = date;
+    }
+}
+
+const currencyRatesArray = Object.entries(currencyRates.rates).map(([code, rate, date]) => new CurrencyRate(code, rate, currencyRates.date));
+
+function insertCurrencyRate() {
+    const base = document.getElementById('base').value.toUpperCase();
+    const targetCurrency = document.getElementById('targetCurrency').value.trim().toUpperCase();
+    const rate = parseFloat(document.getElementById('rate').value);
+    
+    if (!currencyRates) {
+        currencyRates = {
+            timestamp: Date.now(),
+            base: base,
+            date: new Date().toISOString().split('T')[0],
+            rates: {}
+        };
+    }
+
+    if(targetCurrency === '' || rate === ''){
+        alert('Currency or Rate cannot be blank');
+        return;
+    }
+
+    if (currencyRates.base !== base) {
+        alert('Base currency mismatch! Current base is ' + currencyRates.base);
+        return;
+    }
+
+    if(currencyRatesArray.length > 0 && currencyRatesArray.findIndex(obj => obj.code == targetCurrency) !== -1){
+        alert('Curreny already exists in the list');
+        return;
+    }
+
+
+    currencyRates.timestamp = Date.now();
+    currencyRates.date = new Date().toISOString().split('T')[0];
+    currencyRates.rates[targetCurrency] = rate;
+    document.getElementById('newRateOutput').textContent = JSON.stringify(currencyRates, null, 2);
+  // const select = document.getElementById('targetCurrency').textContent;
```

Comment:
Avoid leaving dead code in your PR. I write this comment here, but it also applies to the rest of the file and the other files.